### ISSUE TITLE
Feature: allow manual specification of the ZIP directory structure

### ISSUE THREAD

**Issue by AshesITR** [r-lib/zip#50]
It would be fantastic to allow for manual specification of the ZIP directory structure in a `zip::zipr` or `zip::zipr_append` call, e.g.

```r
zip::zipr("myzip.zip", files = list("myfile.txt" = "myfile-on-disk.txt"))
```

I skimmed the code and think it shouldn't be too hard by changing the `key` in `get_zip_data()` if names are present.
I can give it a try if you think that feature would be useful.

Ideally:

 * for plain files, the name directly translates to the key
 * for directories, the name replaces the key up to the specific directory (so the subdirectories and files have the same relative structure, just a different parent name / path)

**Comment by gaborcsardi** [comment]
Yeah, I think that would be pretty cool. A PR is welcome.

**Comment by AshesITR** [comment]
@gaborcsardi are you still interested in this feature? If so, I can try rebasing #51 onto current `main`.

**Comment by gaborcsardi** [comment]
Yes, sorry for the wait. I'll look at it soon, before the next release.
